packages/web-components/@sfx/sayt/src/sayt.ts

```diff
@@ -77,6 +84,9 @@ export default class Sayt extends LitElement {
     window.addEventListener('click', this.processClick);
     window.addEventListener('keydown', this.processKeyEvent);
     this.setSearchboxListener(this.searchbox, 'add');
+
+    this.requestSaytAutocompleteTerms  = this.debounceSaytRequests(this.requestSaytAutocompleteTerms);
+    this.requestSaytProducts  = this.debounceSaytRequests(this.requestSaytProducts);
```

Comment:
I'm guessing these lines can happen in the constructor, unless I'm missing a reason why you're waiting for `connectedCallback()`.